File: common/src/main/java/com/taskadapter/connector/common/data/ConnectorConverter.java
package com.taskadapter.connector.common.data;

import com.taskadapter.connector.definition.exceptions.ConnectorException;

/**
 * Converter between source and destination type.
 * 
 * @param <S>
 *            source type.
 * @param <T>
 *            destination type.
 */
public interface ConnectorConverter<S, T> {
    /**
     * Converts an inssue from a source to a destination.
     * 
     * @param source
     *            source object to convert.
     * @return converted object.
     * @throws ConnectorException
     *             if object cannot be converted.
     */
    T convert(S source) throws ConnectorException;
}

File: common/src/main/scala/com/taskadapter/connector/common/data/ConnectorConverter.scala
package com.taskadapter.connector.common.data

import com.taskadapter.connector.definition.exceptions.ConnectorException

/**
  * Converter between source and destination type.
  *
  * @tparam S source type.
  * @tparam T destination type.
  */
trait ConnectorConverter[S, T] {
  /**
    * Convert a task from source to target format.
    *
    * @param source source object to convert.
    * @return converted object
    */
  @throws[ConnectorException]
  def convert(source: S): T
}
